src/components/OpportunitiesContent/index.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import { Container, Row, Col } from "react-bootstrap";
+import "./index.scss";
+import { useTranslation } from "react-i18next";
+import OpportunitiesCard from "../OpportunitiesCard";
+import SearchBar from "../SearchBar";
+import JobsTrainingButtons from "../JobsTrainingButtons";
+
+const JobsTrainingContent = () => {
+  const { t } = useTranslation();
+  const [cardsArr, setCardsArr] = useState([
+    ...t(`home.jobOpportunities`, { returnObjects: true }),
+  ]);
+
+  const handleSearch = (e, keyword) => {
+    e.preventDefault();
+    setCardsArr(() => {
+      const searchedCards = [...cardsArr];
```

Comment:
you don't need this line, just filter on previousCards

Reply:
without it, if I search twice, second-time search doesn't appear because it would be searching the filtered array.

Reply:
@louisli

Reply:
@louisrli

Reply:
i'm a bit confused how it searches twice right now. i don't see anything that sets the cardsArr back to the default

* user types "foo"
* setCardsArr -> cardsArr cards contain foo
* user deletes "foo"
* ?? which code sets it back to the original default state of all cards?

Reply:
Const searchedCards = [...cardsArr].

The handle search is filtering the variable searchedCards which always contains all the elements of cardsArr

Reply:
yes, but then you call setCardsArr, so cardsArr has now changed to your new set

i've written searchbox code like this a few times before so to my understanding you are going to need to keep a variable that keeps track of the original set of cards.

the original set of cards is kept in cardsArr. you call setCardsArr which changes cardsArr. therefore you should no longer have access to the original when you call `[...cardsArr]`. does that make sense? i'm only reading the code here, maybe it works for some reason

Reply:
you are reading the code correctly, search twice is not working 😢

Reply:
If I did a copy of the cardsArr like before:
```const [searchCard, setSearchCards] = useState(cardsArr);```
and in handle search I can reset the cards before filterin 
 ```setSearchCards(cardsArr);```

but I'll have to pass it also to JobsTrainingButtons and set both of them to opportunityData
 ```setCardsArr(opportunityData);  ```
 ```setSearchCards(opportunityData) ```

It works like that. I just wanted to make sure, is it best practice or do you suggest is another way?

Reply:
or should I PR what I am suggesting and then you can check?

Reply:
you can do something like that but they need better names. i'll leave it to you to think of it, but you should think about why "cardsarr" and "searchcards" are not expressing what you want them to express if they play those roles.

Reply:
I guess searchCards is not good naming because it doesn't only give the wrong impression but it will be mapped and used even before we trigger search button.

So maybe it can be renderedCards ? Since it is related to the cards appearing on screen whether they are filtered or not.

I am not sure about cardsArr I'll think about why it is not correct the naming. 
But as a first thought maybe it can be selectedOpportunityCards? Since it will always contain and toggle between opportunity data.